### PR TITLE
Fix spec instance get elements

### PR DESCRIPTION
**Pull request by danrubel**
@devoncarew 
I think we need to push the service.md spec change upstream.

+ fix spec Instance elements field type @InstanceRef --> @InstanceRef[]
+ added getInstance utility method
+ enhanced sample and VmServiceTest to test List element access
+ added VmService javadoc

### CODE REVIEW

File: dart/tool/service.md [hunk 1]
```diff
@@ -1368,7 +1368,7 @@ class Instance extends Object {
   //
   // Provided for instance kinds:
   //   List
-  @Instance|Sentinel[] elements [optional];
+  @Instance[]|Sentinel[] elements [optional];
```
Comment (devoncarew) [inline dart/tool/service.md]: Good catch - we should upstream and see what they say.

Note that we are using a slightly older version of service.md, so we should just patch this particular change in.

**Comment by danrubel** [comment]
@devoncarew Fixed the problem and now travis tests pass. Glad you added that travis compile of the java/src directory. It caught the fact that GetInstanceConsumer was in java/test when it should have been in java/src.
PTAL

File: dart/tool/java/generate_java.dart [hunk 2]
```diff
@@ -62,11 +84,16 @@ class Api extends Member with ApiParseUtil {
     _setFileHeader();
 
     // Add undocumented "id" property
+    addProperty('InstanceRef', 'id', javadoc: 'The id of this instance.');
+    addProperty('Instance', 'id', javadoc: 'The id of this instance.');
     addProperty('LibraryRef', 'id', javadoc: 'The id of this library.');
+    addProperty('Library', 'id', javadoc: 'The id of this library.');
     addProperty('ScriptRef', 'id', javadoc: 'The id of this script.');
+    addProperty('Script', 'id', javadoc: 'The id of this script.');
```
Comment (devoncarew) [inline dart/tool/java/generate_java.dart]: `Script` extends `Obj`, which is where it gets it's `id` field. https://github.com/dart-lang/vm_service_drivers/blob/master/dart/lib/vm_service_lib.dart#L1589

The `superName` field is being preserved - perhaps the java generator isn't using it?

**Comment by devoncarew** [comment]
lgtm as is, though we should fix the issue with the superclass inheritance in this PR or the next one

**Comment by danrubel** [comment]
Good point. I'll address that in a subsequent CL. I have a few more CLs queued up first...

File: dart/tool/java/generate_java.dart (hunk 2 above)
Comment (danrubel) [inline dart/tool/java/generate_java.dart]: Great point. I'll address that in a subsequent CL.

**Pull request closed without merging**